### PR TITLE
Feature/wallet manager hook

### PR DESCRIPTION
**Pull request by mrval5**
### What does this do?
We changed the component from class based to hook
### Why are we making this change?
- extending from the class means adding more code in the build
- the lifecycle hooks from the class can be deprecated soon
- the community and React are encouraging to use hooks over class
- most of libraries are using hooks

- testing with Enzyme does not support `useEffect` method, which we have added a new package `jest-react-hooks-shallow`
- We have added also testing-library to help on testing
- we removed some of the unit test related to the component `WalletSelectModal` because this repo is not responsible of testing components from another lib

### Why the PR in draft
- we should merge https://github.com/zer0-os/zOS-component-library/pull/5, and bump it here

### CODE REVIEW

File: src/components/wallet-manager/index.test.tsx [hunk 1]
```diff
@@ -162,46 +133,4 @@ describe('WalletManager', () => {
 
     expect(updateConnector).toHaveBeenCalledWith(Connectors.Metamask);
   });
-
-  describe('mapState', () => {
```
Comment (AdamDotCom) [inline src/components/wallet-manager/index.test.tsx:166]: Where should the state / reducer / action related test coverage be moved to? Should they be move to the ../store directory or would they stay in this file?

File: src/components/wallet-manager/index.tsx [hunk 2]
```diff
@@ -1,116 +1,132 @@
...
-}
-
-export const WalletManager = connectContainer<{}>(Container);
+        </div>
+      )}
+    </div>
+  );
+};
+
+const mapStateToProps = (state: RootState): Partial<Props> => {
```
Comment (AdamDotCom) [inline src/components/wallet-manager/index.tsx:110]: What's the best way to test state mapping? It looks like [redux-mock-store](https://github.com/reduxjs/redux-mock-store) was an option. Is there a more modern alternative?

File: src/components/wallet-manager/index.test.tsx [hunk 3]
```diff
@@ -81,41 +93,25 @@ describe('WalletManager', () => {
...
-    // assert pre-condition
-    expect(wrapper.find(WalletSelectModal).prop('isConnecting')).toBe(true);
-
-    wrapper.setProps({ connectionStatus: ConnectionStatus.Connected });
-
-    // re-open modal, as it will be closed at this point
-    wrapper.find('.wallet-manager__connect-button').simulate('click');
-
-    expect(wrapper.find(WalletSelectModal).prop('isConnecting')).toBe(false);
+    await waitFor(() => {
```
Comment (AdamDotCom) [inline src/components/wallet-manager/index.test.tsx:111]: Cool, I've never encountered `waitFor` before.

**Comment by AdamDotCom** [comment]
@mrval5 how should we attempt to test the state and actions? This article gives a couple options, but the article is two years old: https://jsramblings.com/3-ways-to-test-mapstatetoprops-and-mapdispatchtoprops/
This article demonstrates testing reducers: https://redux.js.org/usage/writing-tests Is there a better way? It'd be nice to see coverage for the state and actions.

File: src/components/wallet-manager/index.test.tsx (hunk 1 above)
Comment (mrval5) [reply src/components/wallet-manager/index.test.tsx:166]: Ideally we should test:
- action creators in store folder (unit test)
- reducers in store folder (unit test)
- testing sagas in store folder (unit test)

and if we want we can test the integrations between the component and the store.

**Review by jtulloch — CHANGES_REQUESTED** [review]
hey! welcome @mrval5. hope you're settling in alright! i know this is a draft, so apologies if any comments are not relevant. feel free to ignore any that are! looking forward to meeting you soon.

**Comment by jtulloch** [comment]
closing this for now until we get a chance to come back to it.

**Pull request closed without merging**